mlir/test/Dialect/MemRef/canonicalize.mlir

```diff
@@ -870,38 +870,6 @@ func.func @reinterpret_of_subview(%arg : memref<?xi8>, %size1: index, %size2: in
 
 // -----
 
-// Check that a reinterpret cast of an equivalent extract strided metadata
```

Comment:
Just looking at these tests... these should still work right?

Reply:
These two tests are testing  a canonicalization feature of `reinterpret_cast` which wouldnt be triggered without the folder of `extract_strided_metadata` so input/output IR are exactly same so I didnt see much value in the test. Should I add it back as is, or should I change it so that the actual interpret cast canonicalization is tested?

Reply:
The real question is: why wasn't the behavior shown in these test the actually desirable behavior of the folder?